src/reports/library/ascii_writer.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+
+#include "ascii_writer.hpp"
+
+AsciiWriter::AsciiWriter(const std::string& report_name)
+: IoWriter(report_name), m_file(report_name) {}
+
+AsciiWriter::~AsciiWriter() {}
+
+void AsciiWriter::write(const std::string& name, float* buffer, int steps_to_write, int total_steps, int total_compartments) {
+    m_file << name << std::endl;
+    size_t buffer_size = total_steps * total_compartments;
+    for(int i=0; i<buffer_size; i++) {
+        m_file << buffer[i] << ", ";
+    }
+    m_file << std::endl << std::endl;
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<int>& buffer) {
+
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<uint32_t>& buffer) {
+    m_file << name << std::endl;
+    for(int elem: buffer) {
+        m_file << elem << ", ";
+    }
+    m_file << std::endl << std::endl;
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<uint64_t>& buffer) {
+    m_file << name << std::endl;
+    for(int elem: buffer) {
+        m_file << elem << ", ";
+    }
+    m_file << std::endl << std::endl;
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<float>& buffer) {
+
+}
+
+void AsciiWriter::write(const std::string& name, const std::vector<double>& buffer) {
+
+}
+
+void AsciiWriter::close() {
```

Comment:
Is this necessary?  Should IOWriter be more RAII?  What is the behavior if the writer is closed, and more data is written to it?

Reply:
That's a good point, I thought about that but it was necessary because coreneuron might call flush() at the end, which will write the remaining steps of the buffer to file and close it.